### ISSUE TITLE
Can linkone link the new item to another item and set relationship?

### ISSUE THREAD

**Issue by gmellis1** [Microsoft/mail2bug#52]
I have a supplier who is using automation to populate bugs into my instance via mail2bug. We want to also set parent-child-predecessor-successor relationships during the email triage transaction as well. Is this possible? If not, we have to manually go in and set this through the interface later on. Thanks

**Comment by hershi** [comment]
There's no support for it currently, though I'd be happy to discuss the exact details of what you're looking (i.e. syntax and logic for deciding on what to link and how)
And, of course, if you're inclined to add the support yourself, don't hesitate to send a PR

**Comment by gxclarke** [comment]
This would be useful for me as well, specifically to link newly created bugs to a fixed parent work item id. Generalizing this a little, the following configuration would support it:

```xml
<WorkItemSettings>
  ...
  <LinkedItems>
    <Link Type="Parent" Target="1473" />
  </LinkedItems>
</WorkItemsSettings>
```

The type attribute should support any allowed link type in TFS, e.g. Parent, Child, Predecessor, Changeset, Commit, Hyperlink.

Is this syntax flexible enough to useful in a broad sense? Is there a need to resolve the target dynamically (although it seems like this would increase the complexity a lot), or to link to a new item that should be created?

**Comment by hershi** [comment]
My assumption is that in most cases, fixing the target (as well as the link type) dynamically. Seems like that's not the case in your specific scenario @gxclarke , but I'm not sure how broadly applicable that scenario is (would be good if you can clarify what's the benefit of having all new bugs associated to the same parent)

Either way, I'd be happy to review a PR for this, but will probably not get to implementing it myself anytime soon

**Comment by gxclarke** [comment]
My case is because we have a practice to log reported bugs under a parent item called "Production Application Support". But I agree that if a solution is implemented, it should be versatile to support a broad range of needs.

One improvement I can think of is to support a dynamic variant on mnemonics that are included in the email, to automatically generate related links:

**`@@@Link(http://something.com)`**

would be the dynamic equivalent to:
```xml
  <Link Type="Hyperlink" Location="http://something.com" />
```
and

**`@@@LinkParent(1473)`**

would be the dynamic equivalent to:
```xml
<Link Type="RelatedLink" LinkTypeEnd="Parent" Target="1473" />
```

Basically, using `@@@Link` on it's own assumes a hyperlink, and using `@@@Link{LinkTypeEnd}` is matched to the applicable LinkTypeEnd. Anything that doesn't match is ignored (with a logged warning).

Thoughts?

**Comment by konpikwastaken** [comment]
Just reiterating what you wrote @gxclarke - it could potentially be something like this:

 1. **`@@@Link(ValidUri)`**
Add Hyperlink to current item.  Shorthand for @@@LinkHyperlink(ValidUri).
Example: **`@@@Link(http://www.github.com)`** would hyperlink "http://www.github.com" to current workitem.

 2. **`@@@Link(ExistingWorkItemId)`**
Add WorkItemId as parent of current work item.  Shorthand for @@@LinkParent(ExistingWorkItemId).
Example: **`@@@Link(123)`** would link 123 as the parent of the current workitem.

 3. **`@@@LinkLinkTypeEnd(ExistingWorkItemId)`**
Add `LinkTypeEnd` link type with `ExistingWorkItemId` at the end. 
Example 1: **`@@@LinkParent(123)`** would link 123 as the _parent_ of the current workitem.
Example 2: **`@@@LinkRelated(345)`** would link 345 as the _related_ item of the current workitem (non-hierarchy type links)

Would the config driven setting be always applied in your scenario? How about supporting link comments, (e.g. `@@@Link(ValidUri)(My comment for this link could go here)`)? Any value there?
